ibmdbpy4nps/base.py

```diff
@@ -175,38 +177,62 @@ def __init__(self, dsn, uid='', pwd='', hostname = '', port = 0, autocommit=True
         # first delimiter before the parameters in the jdbc-url
         url_1stparam_del = ':'
 
+
         # Detect if user attempt to connection with ODBC or JDBC
-        if dsn.startswith('jdbc:'):
-            self._con_type = "jdbc"
-            if dsn.startswith('jdbc:netezza:'):
-                self._database_system = 'netezza'
-                # for Netezza the internal connection is always in autocommit mode
-                # to allow explicit commits
-                dsn+= ';autocommit=false'
-                url_1stparam_del = ';'
-            elif dsn.startswith('jdbc:db2:'):
-                self._database_system = 'db2'
+        if isinstance(dsn,str):
+            if dsn.startswith('jdbc:'):
+              self._con_type = "jdbc"
+              if dsn.startswith('jdbc:netezza:'):
+                  self._database_system = 'netezza'
+                  # for Netezza the internal connection is always in autocommit mode
+                  # to allow explicit commits
+                  dsn += ';autocommit=false'
+                  url_1stparam_del = ';'
+              elif dsn.startswith('jdbc:db2:'):
+                  self._database_system = 'db2'
+              else:
+                  raise IdaDataBaseError(("The JDBC connection string is invalid for Db2 and Netezza. " +
+                                          "It has to start either with 'jdbc:db2:' or 'jdbc:netezza:'."))
+
             else:
-                raise IdaDataBaseError(("The JDBC connection string is invalid for Db2 and Netezza. " +
-                                        "It has to start either with 'jdbc:db2:' or 'jdbc:netezza:'."))
-        elif hostname != '':
+                self._con_type='odbc'
+
+        elif isinstance(dsn, dict):
             self._con_type = "nzpy"
-        else:
-            self._con_type = "odbc"
+
 
         self._idadfs = []
 
+
+
         if self._con_type == 'nzpy':
+            #parse dict
+            host=''
+            port=0
+            database=''
+            securityLevel=0
+            for key,value in dsn.items():
+                if(key=='host'):
+                    host=value
+                if (key=='port'):
+                    port=value
+                if (key=='database'):
+                    database=value
+                if (key=='securityLevel') :
+                    securityLevel=value
+                if (key=='logLevel') :
+                    logLevel=value
```

Comment:
user and password should be added as well, because they are parameters of nzpy.connect. They should be treated in the same way as for jdbc-urls which can include user and passwords as well.